pulsar/connector/producer.go

```diff
@@ -9,16 +9,14 @@ import (
 	"go.uber.org/zap"
 )
 
-func CreateProducer(pulsarClient pulsar.Client, topic string) (pulsar.Producer, error) {
+func CreateProducer(pulsarClient pulsar.Client, topic TopicName) (pulsar.Producer, error) {
 	// Get a producer instance
 	producer, err := pulsarClient.CreateProducer(pulsar.ProducerOptions{
-		Topic: topic,
+		Topic: GetTopic(topic),
```

Comment:
again, GetTopic should be a method of the client

Reply:
Removed this method 
user needs to supply namespace and tenant - if not specified it is taken from the global config